src/unix/fs.c

```diff
@@ -860,7 +860,11 @@ static ssize_t uv__fs_buf_iter(uv_fs_t* req, uv__fs_buf_iter_processor process)
 
   if (bufs != req->bufsml)
     uv__free(bufs);
-  req->bufs = NULL;
+
+  if (!(errno == EINTR && total == -1)) {
```

Comment:
We are retrying only if `retry_on_eintr` is non-zero. Do you think it may be beneficial to pass it to this function and check it in this condition?